### PR TITLE
Add dns trace test type

### PR DESCRIPTION
**Pull request by johntdyer**

### CODE REVIEW

File: dns_trace.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	TestID               int            `json:"testId,omitempty"`
+	TestName             string         `json:"testName,omitempty"`
+	Type                 string         `json:"type,omitempty"`
+	DNSTransportProtocol string         `json:"dnsTransportProtocol,omitempty"`
+	Domain               string         `json:"domain,omitempty"`
+	Interval             int            `json:"interval,omitempty"`
+}
+
+// AddDNSTrace - Add dns trace test
+func (t *DNSTrace) AddDNSTrace(id int) {
```
Comment (william20111) [inline dns_trace.go:32]: AddAgent ?

File: dns_trace.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+//DeleteDNSTrace - delete dns trace test
+func (c *Client) DeleteDNSTrace(id int) error {
+	resp, err := c.post(fmt.Sprintf("/tests/dns-trace/%d/delete", id), nil, nil)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != 204 {
+		return fmt.Errorf("failed to delete page load, response code %d", resp.StatusCode)
```
Comment (william20111) [inline dns_trace.go:73]: error msg update

File: dns_trace.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	}
+	return nil
+}
+
+//UpdateDNSTrace - - delete dns trace test
```
Comment (william20111) [inline dns_trace.go:78]: comment for update

File: dns_trace.go (hunk 1 above)
Comment (johntdyer) [reply dns_trace.go:32]: yea, I am adding just test types here.  FWIW I think the test vs agent model of the API is going to lead to confusion w/ this SDK... Especially w/o including `Test` in the name of the tests... Not sure how you feel about that...

Comment (johntdyer) [reply dns_trace.go:32]: image

Comment (johntdyer) [reply dns_trace.go:32]: @william20111 you can see there is a type of test called "dns-trace"

**Pull request merged**